src/ray/raylet/worker_pool.cc

```diff
@@ -240,13 +272,20 @@ Process WorkerPool::StartWorkerProcess(const Language &language,
         << " placeholder is not found in worker command.";
   }
 
+  // TODO(kfstorm): Set up environment variables in a later PR.
   Process proc = StartProcess(worker_command_args);
+  if (RayConfig::instance().enable_multi_tenancy()) {
+    // If the pid is reused between processes, the old process must have exited.
```

Comment:
`ProcessDisconnectClientMessage` will be invoked if the worker process is dead because the RPC connection is broken. But it has nothing to do with this comment. The comment is talking about what if a newly created worker process has the PID which is also used by a historical worker process probably a long time ago. In this case, the old worker process must have already been dead. So it's safe to update the value of the <pid, job id> map.